### ISSUE TITLE
reportCallIssue error when making instance from Annotated type(version 1.1.348 only)

### ISSUE THREAD

**Issue by cbscsm** [microsoft/pyright#7106]
When I updated 1.1.348 from 1.1.347, pydantic's Annotated constructors causes type errors with pyright.
The minimized reproducing code:
```python
from typing import Annotated

String = Annotated[str, str]

s: str = String(3)
```
Error:
```
Object of type "type[str]" is not callable  (reportCallIssue)
Type of "s" is partially unknown
  Type of "s" is "Unknown | str"  (reportUnknownVariableType)
```
I tested it on [pyright playground](https://pyright-play.net/?pyrightVersion=1.1.348&strict=true&locale=en&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoCCKKYMAhjAKYAmAULQMowioYC8hxpFNA2gM7MANFEEgAuvX4AuUcygcmLdAAoAzAEpaQA).

On pyright version 1.1.347 or mypy, type checking is passed.

**Comment by erictraut** [comment]
This is a duplicate of https://github.com/microsoft/pyright/issues/6875. A final decision hasn't yet been made on this, but it's likely that you will need to change your code here. You're using `Annotated` in a way that isn't valid — or is at best ambiguous and not anticipated by the spec.

**Comment by sdherr** [comment]
Could you please clarify why this isn't valid? According to the [stdlib docs](https://docs.python.org/3/library/typing.html#typing.Annotated), `Annotated` is _supposed_ to be for adding metadata to a type and having the resulting type behave exactly like the pre-annotated type.

```
If a library or tool encounters an annotation Annotated[T, x] and has no special logic for the metadata, it should ignore the metadata and simply treat the annotation as T. As such, Annotated can be useful for code that wants to use annotations for purposes outside Python’s static typing system.

Using Annotated[T, x] as an annotation still allows for static typechecking of T, as type checkers will simply ignore the metadata x. In this way, Annotated differs from the [@no_type_check](https://docs.python.org/3/library/typing.html#typing.no_type_check) decorator, which can also be used for adding annotations outside the scope of the typing system, but completely disables typechecking for a function or class.
```

`str` _is_ callable, so it seems to me that `String` in the above example should be recognized as callable too.

I have this issue when using [Annotated FastAPI Dependencies](https://fastapi.tiangolo.com/tutorial/dependencies/#share-annotated-dependencies), but the root problem is the same, pyright doesn't recognize the annotated type as callable even though the original type is. I suppose I could always _also_ import the original type and instantiate it directly when I need to, but it seems like this should work.

**Comment by erictraut** [comment]
Please refer to [this discussion in the typing forum](https://discuss.python.org/t/is-annotated-compatible-with-type-t/43898/14).
